src/popup/searchModule.js

```diff
@@ -105,8 +105,9 @@ function appendToGrid(msnry, fragment, divs, grid) {
 }
 
 // TODO: be more specific
+// Object.prototype.hasOwnProperty.call(apiResponse, 'validation_error'
 export function checkValidationError(apiResponse) {
-  if (Object.prototype.hasOwnProperty.call(apiResponse, 'validation_error')) {
+  if (apiResponse.result_count === 0) {
```

Comment:
As mentioned in the issue, we have to adapt to the new response during any error. As you can see here (https://api.creativecommons.engineering/v1/images?q=+), there is no key `result_count`. Therefore this condition should not belong here. And this also has a negative side effect, because in case the user enters a query which does not have any images, (eg: https://api.creativecommons.engineering/v1/images?q=sdfjsdfksajrfaskdfas) then it should have been handled by [checkResultLength()](https://github.com/creativecommons/ccsearch-browser-extension/blob/master/src/popup/popup.js#L276) function and a message of `No images found. Please enter a different query` must have been show, but in this implementation `checkValidationError` function will block it.

Reply:
I didn't understand how I should check it. So since the new v1 api doesn't have validation_error as property instead check if the response has a error_type .

Reply:
For this PR, it would be better that you remove this condition block.